qiskit/synthesis/unitary/qsd.py

```diff
@@ -106,6 +107,13 @@ def qs_decomposition(
            n-Qubit Gates Based on Block ZXZ-Decomposition*,
            `arXiv:2403.13692 <https://arxiv.org/abs/2403.13692>`_
     """
+    if decomposer_1q is None and decomposer_2q is None:
+        array = np.asarray(mat, dtype=complex)
+        return QuantumCircuit._from_circuit_data(qsd.qs_decomposition(array, opt_a1, opt_a2))
+
+    # TODO call rust if decomposer_1q or decomposer_2q are OneQubitEulerDecomposer or
+    #  TwoQubitBasisDecomposer, deprecate otherwise
```

Comment:
We should do this still we can save the deprecation for a follow on, but if we believe there is value in allowing users to provide their own 1q and 2q decomposer for use internally we should pass the objects to the inner rust function if their of the correct type.

Reply:
Is this actionable in this PR or should we wait for a future update?